src/Console/Application.php

```diff
@@ -130,4 +144,68 @@ private function createContext(InputInterface $input, OutputInterface $output):
 
         return $descriptor->function->invoke(...$args);
     }
+
+    private function displayUpdateWarningIfNeeded(SymfonyStyle $symfonyStyle): void
+    {
+        $latestVersion = $this->cache->get('latest-version', function (ItemInterface $item): array {
+            $item->expiresAfter(1);//3600 * 60);
+            $opts = [
+                "http" => [
+                    "method" => "GET",
+                    "header" => "User-Agent: castor\r\n",
+                ]
+            ];
+
+            $context = stream_context_create($opts);
+
+            $content = @file_get_contents('https://api.github.com/repos/jolicode/castor/releases/latest', false, $context);
+
+            return json_decode($content, true) ?? [];
+        });
+
+        if (!$latestVersion) {
+            log('Failed to fetch latest Castor version from GitHub.', 'warning');
+
+            return;
+        }
+
+        if (version_compare($latestVersion['tag_name'], Application::VERSION, '<=')) {
+            return;
+        }
```

Comment:
This is something we discussed internally but we are not fixed on this yet. But I think once v1 is released, the API will be stable for a long time, so I would say yes, it will be safe in most cases. And it should always be simple to rollback to a previous version if needed IMO